pkg/apis/volumesnapshot/v1beta1/types.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package v1alpha1
+package v1beta1
 
 import (
 	core_v1 "k8s.io/api/core/v1"
```

Comment:
In VolumeSnapshotContent:
```
	// Optional: what happens to a snapshot content when released from its snapshot. It will be set to Delete by default
	// if not specified		// if not specified
	// +optional		// +optional
	DeletionPolicy *DeletionPolicy `json:"deletionPolicy" protobuf:"bytes,5,opt,name=deletionPolicy"`		DeletionPolicy *DeletionPolicy `json:"deletionPolicy" protobuf:"bytes,5,opt,name=deletionPolicy"`
```

The default should be retain. If this object is created and this is not set it should be "retain" -- "principle of least surprise".

Reply:
P0

Reply:
my understanding is it is possible that this field is not set (admission controller will not defaulting it, and the behavior of the controller should be retain the snapshot content.